lib/index.js

```diff
@@ -10,23 +10,41 @@ const Joi = require('joi');
 const Schema = require('./schema');
 const Package = require('../package.json');
 
-// const SchwiftyMigration = require('schwifty-migration');
-
 const internals = {};
 
 exports.register = function (server, options, next) {
 
+    Joi.assert(options, Schema.plugin, 'Bad plugin options passed to schwifty.');
+
     const rootState = internals.state(server.root);
 
-    internals.state(server).dir = options.dir;
     const knexFile = options.knexFile;
 
-    internals.state(server).knex = Knex(knexFile[process.env.NODE_ENV]);
+    /*
+        Move the models into the knexFile object
+        Since a models path isn't part of it's schema on knexjs.org
+    */
+
+    if (options.models) {
+        knexFile.models = options.models;
+    }
+
+    let knex;
+
+    try {
+        knex = Knex(knexFile[process.env.NODE_ENV]);
+    }
+    catch (err) {
```

Comment:
I don't think this actually needs a try-catch unless there are some _specific_ errors you'd like to handle.  As written I think this risks mishandling programmer-errors.  Check this out if you haven't already seen it!  https://www.joyent.com/node-js/production/design/errors

If there were an operational error, like if something went wrong connecting to the db, it would probably happen asynchronously and you wouldn't be able to catch it.